Sources/Sentry/SentryProfiler.mm

```diff
@@ -458,6 +466,14 @@ - (void)stopForReason:(SentryProfilerTruncationReason)reason
         return;
     }
 
+#    if SENTRY_HAS_UIKIT
+    // if SentryOptions.enableAutoPerformanceTracing is NO, then we need to stop the frames tracker
+    // from running outside of profiles because it isn't needed for anything else
+    if (![[[[SentrySDK currentHub] getClient] options] enableAutoPerformanceTracing]) {
```

Comment:
l: We should find a better way to get the options sentry was initialized with. 
But this is a thing for another PR.